alg/crc32c_sse42.c

```diff
@@ -0,0 +1,31 @@
+#include "cpusupport.h"
+#ifdef CPUSUPPORT_X86_CRC32
+
+#include <smmintrin.h>
+
+#include "crc32c_sse42.h"
+
+/**
+ * CRC32C_Update_SSE42(state, buf, len):
+ * Feed ${len} bytes from the buffer ${buf} into the CRC32C whose state is
+ * ${state}.  This implentation uses x86 SSE4.2 instructions, and should only
+ * be used if CPUSUPPORT_X86_CRC32 is defined and cpusupport_x86_crc32 returns
+ * nonzero.
+ */
+void
+CRC32C_Update_SSE42(uint32_t * state, const uint8_t * buf, size_t len)
```

Comment:
Question about naming this function, and the source file in general.  Following the pattern of other CPU-specific source files, we'd have `crc32c_crc32` which IMO looks goofy.  I went with `crc32c_SSE42`, but I'll be entirely unsurprised if you tell me to switch to `crc32c_crc32`.